viewing_party/main.py

```diff
@@ -0,0 +1,153 @@
+
+def get_most_watched_genre(user_data):
+    genres = {}
+    watchedlist = user_data["watched"]
+    if len(watchedlist) == 0:
+        return None
+    else:
+        for i in range(len(watchedlist)):
+            if watchedlist[i]["genre"] not in genres:
+                genres[watchedlist[i]["genre"]] = 1
+            else:
+                genres[watchedlist[i]["genre"]] += 1
+    sort_genres = sorted(genres.items(), key=lambda x: x[1], reverse=True)
+    return sort_genres[0][0]
+
+def get_unique_watched(user_data):
+    userwatched = user_data["watched"]
+    friendswatched = user_data["friends"]
+    unique_watched = []
+    friendtitlelist = []
+    for friend in friendswatched:
+        watched = friend["watched"]
+        for title in watched:
+            friendtitlelist.append(title["title"])
+    for title in userwatched:
+        if title["title"] not in friendtitlelist:
+            unique_watched.append(title)
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    userwatched = user_data["watched"]
+    friendswatched = user_data["friends"]
+    userlist = []
+    unique_watched = []
+    for title in userwatched:
+        userlist.append(title["title"])
+    for friend in friendswatched:
+        watched = friend["watched"]
+        for title in watched:
+            if title["title"] not in userlist:
+                if title not in unique_watched:
+                    unique_watched.append(title)
+    print(unique_watched)
+    return unique_watched
+
+
+def get_available_recs(user_data):
+    user_hosts_list = user_data["subscriptions"]
+    user_watched_list = user_data["watched"]
+    friends_dictkey_watched_list_of_dicts = user_data["friends"]
+    users_list_of_watched_titles = []
+    recommendations = []
+    for title in user_watched_list:
+        users_list_of_watched_titles.append(title["title"])
+    for friend in friends_dictkey_watched_list_of_dicts:
+        friends_watched_list_of_dicts = friend["watched"]
+        for movie in friends_watched_list_of_dicts:
+            if movie["title"] not in users_list_of_watched_titles:
+                if movie["host"] in user_hosts_list:
+                    if movie not in recommendations:
+                        recommendations.append(movie)
+    
+    return recommendations 
+
+
+def get_new_rec_by_genre(user_data):
+    most_watched_genre = get_most_watched_genre(user_data)
+    user_watched_list = user_data["watched"]
+    friends_dictkey_watched_list_of_dicts = user_data["friends"]
+    users_list_of_watched_titles = []
+    recommendations = []
+    for title in user_watched_list:
+        users_list_of_watched_titles.append(title["title"])
+    for friend in friends_dictkey_watched_list_of_dicts:
+        friends_watched_list_of_dicts = friend["watched"]
+        for movie in friends_watched_list_of_dicts:
+            if movie["title"] not in users_list_of_watched_titles:
+                if movie["genre"] is most_watched_genre:
```

Comment:
In this conditional statement use `==`.  For an explanation of is vs ==: https://www.geeksforgeeks.org/difference-operator-python/